Code issues reported by PVS addressed

BjossiAlfreds opened the pull request:
This pull request fixes (almost) all of the issues reported in bug https://github.com/yquake2/xatrix/issues/17.

The exceptions are:
1. The type-punning stuff in g_svcmds and shared I did not mess with. Instead I added a comment for future reference, suggesting using memcpy which is, as far as I know, the correct way to do type-punning with compiler optimizations enabled.

2. The PVS analysis does not know that gi.error() doesn't return so it falsely reported a number of issues in savegame.c. Perhaps gi.error() should be decorated with _Noreturn (one of the newer features of the C standard)?

3. Short global var names in rand.c. They are declared static and are in a file not likely to change in the future so I don't see this as an issue. But feel free to give them longer names if you want.

The rest of the issues I did address were a variety of dead code elimination, expression and block simplification, some possible null pointer refs, incorrect use of rand() instead of random(), etc. Overall not many outright bugs were fixed but there were a few in there.

Yamagi commented:
I'll look into this (and likely merge it) but it may take another week. I'm currently somewhat short on time. So far we haven' started with Ground Zero. But there's ongoing work for the main game: https://github.com/yquake2/yquake2/tree/pvs-studio-fixes

BjossiAlfreds commented:
There's no hurry, the game seems fairly stable as it is. I don't think I introduced any new issues but do review my changes to make sure before you merge. Most of these were redundant expressions that I removed for simplification, like checking a pointer for NULL twice.

Yamagi commented:
I've finally found the time to review and test it. I'm fine with this and I'll merged it as it is. Just a small nit: Can you rebase the branch onto master to remove the merge commits?
